spec/unit/functions/reverse_each_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'puppet'
+require 'spec_helper'
+require 'puppet_spec/compiler'
+
+require 'shared_behaviours/iterative_functions'
+
+describe 'the reverse_each method' do
+  include PuppetSpec::Compiler
+
+  it 'raises an error when used against an unsupported type' do
+    expect do
+      compile_to_catalog(<<-MANIFEST)
+        3.14.reverse_each |$v| {  }
+      MANIFEST
+    end.to raise_error(Puppet::Error, /Cannot be coerced into an Iterable/)
+  end
+
+  it 'raises an error when called with any parameters besides a block' do
+    expect do
+      compile_to_catalog(<<-MANIFEST)
+        [1].reverse_each(1) |$v| {  }
+      MANIFEST
+    end.to raise_error(Puppet::Error, /expects 1 argument, got 2/)
+  end
+
+  it 'raises an error when called with something that is not a block' do
+    expect do
+      compile_to_catalog(<<-MANIFEST)
+        [1].reverse_each(1)
+      MANIFEST
+    end.to raise_error(Puppet::Error, /expects 1 argument, got 2/)
```

Comment:
This test looks strage - it fails on getting two arguments (ike example above), not on being given something that is not a block) - you cannot really do that in puppet source as the only way to get a block there is to specify one.